sys/net/gnrc/link_layer/s_cosens_node/s_cosens_node.c

```diff
@@ -0,0 +1,681 @@
+                reply.content.value = (uint32_t) ok;
+                msg_reply(&msg, &reply);
+                break;
+
+            /* get specific option */
+            case GNRC_NETAPI_MSG_TYPE_SET:
+                /* TODO: filter out MAC layer options -> for now forward
+                         everything to the device driver */
+                DEBUG("s_cosens_node: GNRC_NETAPI_MSG_TYPE_SET received\n");
+                /* read incoming options */
+                opt = (gnrc_netapi_opt_t *)msg.content.ptr;
+                /* set option for device driver */
+                res = dev->driver->set(dev, opt->opt, opt->data, opt->data_len);
+                DEBUG("s_cosens_node: response of netdev->set: %i\n", res);
+                /* send reply to calling thread */
+                reply.type = GNRC_NETAPI_MSG_TYPE_ACK;
+                reply.content.value = (uint32_t)res;
+                msg_reply(&msg, &reply);
+                break;
+
+            /* set specific option */
+            case GNRC_NETAPI_MSG_TYPE_GET:
+                /* TODO: filter out MAC layer options -> for now forward
+                         everything to the device driver */
+                DEBUG("s_cosens_node: GNRC_NETAPI_MSG_TYPE_GET received\n");
+                /* read incoming options */
+                opt = (gnrc_netapi_opt_t *)msg.content.ptr;
+                /* get option from device driver */
+                res = dev->driver->get(dev, opt->opt, opt->data, opt->data_len);
+                DEBUG("s_cosens_node: response of netdev->get: %i\n", res);
+                /* send reply to calling thread */
+                reply.type = GNRC_NETAPI_MSG_TYPE_ACK;
+                reply.content.value = (uint32_t)res;
+                msg_reply(&msg, &reply);
+                break;
+
+            /* unknown message: crash the program */
+            default:
+                sprintf(errmsg,
+                        "s_cosens_node: Unknown command %" PRIu16 "\n",
+                        msg.type);
+                core_panic(PANIC_ASSERT_FAIL, errmsg);
+                /* won't go past here */
+                break;
+        }
+    }
+    /* never reached */
+    return NULL;
+}
+
+
+/****************************************************************************/
+/*    Initialization function exported by the S-CoSenS protocol's driver    */
+/****************************************************************************/
+
+kernel_pid_t gnrc_s_cosens_node_init(char *stack,
+                                     int stacksize,
+                                     char priority,
+                                     const char *name,
+                                     gnrc_netdev_t *dev)
+{
+    /* define the CSMA/CA parameters */
+    set_csma_mac_min_be(S_COSENS_NODE_MAC_MIN_BE);
+    set_csma_mac_max_be(S_COSENS_NODE_MAC_MAX_BE);
+    set_csma_mac_max_csma_backoffs(S_COSENS_NODE_MAC_MAX_CSMA_BACKOFFS);
```

Comment:
What if one runs two interfaces, one with S-CoSenS node configuration and one with S-CoSenS router configuration (or one with CoSenS from #4180). Then the setting of these might lead to a problem.